src/components/Clock/Clock.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import './Clock.scss';
+
+type Props = {
+  name: string,
+};
+
+type State = {
+  date: Date;
+  timerId: NodeJS.Timer
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    date: new Date(),
+    timerId: setInterval(() => {}, 0),
+  };
+
+  componentDidMount() {
+    this.state.timerId = setInterval(() => {
+      this.setState({ date: new Date() });
+
+      // eslint-disable-next-line
+      console.log(this.state.date.toLocaleTimeString());
```

Comment:
setState is an async function, u need to create const with time and use him in setState and console.log